src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/auditing/AuditableEvent.kt

```diff
@@ -21,8 +21,12 @@ data class AuditableEvent(
       type: AuditEventType,
       timeSource: TimeSource,
       authentication: Authentication?,
-      metadata: Map<String, Any>? = null
-    ): AuditableEvent {
+      metadata: Map<String, Any>? = null,
+      allowNoUser: Boolean = false
+    ): AuditableEvent? {
```

Comment:
As discussed we should not really be able to have nullable AuditableEvents.  It can only be null in exceptional circumstances in which case we should just throw a runtime exception.